pkg/global/upgrade.go

```diff
@@ -100,8 +116,21 @@ func Upgrade(version string) (string, error) {
 	return version, nil
 }
 
-func untar(reader io.Reader, target string) error {
-	tarReader := tar.NewReader(reader)
+func untar(reader io.Reader, target string, totalSize int64) error {
+	p := mpb.New(mpb.WithWidth(64))
+	bar := p.AddBar(totalSize,
```

Comment:
we do not want this function printing anything out to the terminal - everything in `pkg` should be agnostic of where it's being called from

to support a progress thing like this the `Upgrade` command would have to take a callback or something

Reply:
@thdxr I guess https://github.com/sst/ion/commit/9ae9d3d1c972cdd4318b9f8d54e2c15814f4edcf after this pr can closed?